### ISSUE TITLE
AuthServiceProvider User type mismatch using existing Laravel project

### ISSUE THREAD

**Issue by keithk** [waterholeforum/core#53]
Thank you for the update that adds existing laravel project integration 🙏

### Bug description

Error:
```
Waterhole\Providers\AuthServiceProvider::Waterhole\Providers\{closure}(): Argument #1 ($user) must be of type Waterhole\Models\User, App\Models\User given
```

Here is the [flare trace](https://flareapp.io/share/J7oxrl35).

### Steps to reproduce

- Followed the documentation for [Laravel Integration ](https://waterhole.dev/docs/main/laravel) through to the end. This error happens on the first load, when redirected to `forum/register/X`

### Environment

```yaml
Environment ......................................................................................................................................  
  Application Name ......................................................................................................................... 🤖 Kumos  
  Laravel Version .......................................................................................................................... 10.38.2  
  PHP Version ............................................................................................................................... 8.2.14  
  Composer Version ........................................................................................................................... 2.5.5  
  Environment ................................................................................................................................ local  
  Debug Mode ............................................................................................................................... ENABLED  
  URL ................................................................................................................................... kumos.test  
  Maintenance Mode ............................................................................................................................. OFF  
  Waterhole Version .......................................................................................................................... 0.4.1
```

**Comment by tobyzerner** [comment]
Thanks for the report! This looks like a bug, I'll aim to do a patch release with a fix in the next couple days.

**Comment by tobyzerner** [comment]
Hi @keithk, could you please update to waterhole/core dev-main and see if the latest commit fixes your issue?

**Comment by keithk** [comment]
@tobyzerner We're getting closer! I could see it do the redirect and then was greeted with a new error. Here is the [shared flare error](https://flareapp.io/share/bP91jMOP/claim/G6tZmfsxrshRB42hv1Wty583NzXBIyU8).

Now we've got:

```
Spatie\Permission\PermissionRegistrar::Spatie\Permission\{closure}(): Argument #1 ($user) must be of type Illuminate\Contracts\Auth\Access\Authorizable, Waterhole\Models\User given, called in /Users/keith/Herd/kumos/vendor/laravel/framework/src/Illuminate/Auth/Access/Gate.php on line 566
```

**Comment by tobyzerner** [comment]
@keithk Thanks for the info! I've just pushed another commit to add the Authorizable contract to Waterhole's User model - can you try updating again and let me know how it goes?

**Comment by keithk** [comment]
@tobyzerner stunning! worked like a charm. thanks for the debugging over the holidays, I can't wait to dig in! 

<img width="1179" alt="Screenshot 2023-12-28 at 6 07 58 PM" src="https://github.com/waterholeforum/core/assets/68270/be07d47c-6338-4cef-b199-60b453f01b84">

**Comment by tobyzerner** [comment]
Awesome @keithk, let me know if you run into any further issues! I've released this fix in 0.4.2.
